### PR TITLE
GFN level now accessible

### LINKED ISSUE #33: Make GFN level accessible

### PR DESCRIPTION
**Pull request by jonathan-schoeps**
closes #33 
The GFN-level is now accessible via the .toml file or via a flag.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -14,6 +14,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Shifted group / element sorting definitions to miscellaneous
 - `xyz` files are written on the fly, and not post-generation
 - `forbidden_elements` and `element_composition` influences hydrogen and organic element addition
+- GFN level now possible to set
```
Comment (marcelmbn) [inline CHANGELOG.md:17]:
```suggestion
- GFN<n>-xTB level can now be set
```

File: mindlessgen.toml [hunk 2]
```diff
@@ -65,6 +65,8 @@ debug = false
 [xtb]
 # > Path to the xtb executable. The names `xtb` and `xtb_dev` are automatically searched for. Options: <str | Path>
 xtb_path = "/path/to/xtb"
+# > The GFN<n>-xTB method that should be used:Options: <int = 0> = GFN-0, <int = 1> = GFN-1, <int = 2> = GFN-2
```
Comment (marcelmbn) [inline mindlessgen.toml:68]:
```suggestion
# > The GFN<n>-xTB method that should be used: Options: <int = 0> -> GFN-0-xTB, <int = 1> -> GFN-1-xTB, <int = 2> -> GFN-2-xTB
```

File: src/mindlessgen/prog/config.py [hunk 3]
```diff
@@ -560,6 +561,22 @@ def xtb_path(self, xtb_path: str | Path):
...
+        """
+        return self._level
+
+    @level.setter
+    def level(self, level: int):
+        """
+        Set the refinement level.
+        """
+        if not isinstance(level, int):
+            raise TypeError("Refinement level should be a string.")
```
Comment (marcelmbn) [inline src/mindlessgen/prog/config.py:577]:
```suggestion
            raise TypeError("xtb level level should be an integer.")
```

Probably copied and not adjusted ;)

File: src/mindlessgen/prog/config.py [hunk 4]
```diff
@@ -560,6 +561,22 @@ def xtb_path(self, xtb_path: str | Path):
...
+    def level(self):
+        """
+        Get the refinement level.
+        """
+        return self._level
+
+    @level.setter
+    def level(self, level: int):
+        """
+        Set the refinement level.
```
Comment (marcelmbn) [inline src/mindlessgen/prog/config.py:574]:
```suggestion
        Set the GFN<n>-xTB level.
```

File: src/mindlessgen/prog/config.py [hunk 5]
```diff
@@ -560,6 +561,22 @@ def xtb_path(self, xtb_path: str | Path):
             raise TypeError("xtb_path should be a string.")
         self._xtb_path = xtb_path
 
+    @property
+    def level(self):
+        """
+        Get the refinement level.
```
Comment (marcelmbn) [inline src/mindlessgen/prog/config.py:567]:
```suggestion
        Get the GFN<n>-xTB level.
```

File: src/mindlessgen/qm/xtb.py [hunk 6]
```diff
@@ -42,11 +42,13 @@ def optimize(
             molecule.write_xyz_to_file(str(temp_path / "molecule.xyz"))
 
             # run xtb
+            gfn_level = self.cfg.level
```
Comment (marcelmbn) [inline src/mindlessgen/qm/xtb.py:45]: Why do you initialize a single variable beforehand? Is this necessary? Why not putting `self.cfg.level` directly in the `arguments` definition?

File: src/mindlessgen/qm/xtb.py [hunk 7]
```diff
@@ -86,10 +88,12 @@ def singlepoint(self, molecule: Molecule, verbosity: int = 1) -> str:
...
```
Comment (marcelmbn) [inline src/mindlessgen/qm/xtb.py:91]: See above.

**Review by marcelmbn — COMMENTED** [review]
After including all suggestions and resolving all questions, this should be good to go.

File: src/mindlessgen/prog/config.py (hunk 3 above)
Comment (jonathan-schoeps) [reply src/mindlessgen/prog/config.py:577]: is the text now "xtb level level should be an integer? ", at least this stands in the changed line

File: src/mindlessgen/qm/xtb.py (hunk 6 above)
Comment (jonathan-schoeps) [reply src/mindlessgen/qm/xtb.py:45]: To have the variable at hand in case I needed it later.

Comment (marcelmbn) [reply src/mindlessgen/qm/xtb.py:45]: But you have the variable also as an attribute from `self.cfg` at hand? There's no advantage of assigning this to another variable.

File: src/mindlessgen/prog/config.py (hunk 3 above)
Comment (marcelmbn) [reply src/mindlessgen/prog/config.py:577]: Don't understand the question – if you commit the suggestion, then yes; currently not. 😄

Comment (marcelmbn) [reply src/mindlessgen/prog/config.py:577]: Good catch, should be only a single "level" obviously.

File: src/mindlessgen/qm/xtb.py (hunk 6 above)
Comment (jonathan-schoeps) [reply src/mindlessgen/qm/xtb.py]: The changes should now be included.

**Review by marcelmbn — APPROVED** [review]
LGTM.

**Pull request merged**